units/tests/serde.rs

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Test the `serde` implementations for types in `units`.
+
+#![cfg(feature = "serde")]
```

Comment:
```suggestion
#![cfg(all(feature = "serde", feature = "alloc"))]
```
`bitcoin_units::amount::serde::as_btc` is behind an `alloc` feature gate causing the CI failure

Reply:
I put them separately

```rust
#![cfg(feature = "alloc")]
#![cfg(feature = "serde")]
```